ui/src/app/kvm/views/KVMDetails/CoreResources/CoreResources.tsx

```diff
@@ -1,16 +1,28 @@
 import classNames from "classnames";
 
 import PodMeter from "app/kvm/components/PodMeter";
+import { getRanges } from "app/utils";
 
 export type Props = {
   cores: {
     allocated: number;
     free: number;
   };
   dynamicLayout?: boolean;
+  pinned?: number[];
+  available?: number[];
 };
 
-const CoreResources = ({ cores, dynamicLayout }: Props): JSX.Element => {
+const CoreResources = ({
+  cores,
+  dynamicLayout,
+  pinned = [],
+  available = [],
+}: Props): JSX.Element => {
+  const noPin = (pinned && pinned.length === 0) || 0;
```

Comment:
The `no` variables are a liiiittle bit strange because they mix types (boolean/number) and check for the presence of `length` even though a default is set.

Anyway I think all we really want to determine is whether or not to show the "Pinned cores" section, which should only be if any `available` or `pinned` are provided. You could do this in many different ways, but as an example (with `available` and `pinned` set to `[]` by default):
``` tsx
const showPinnedSection = available.length + pinned.length > 0;
```

Reply:
I like this solution, so much easier 😄